src/vtfpp/VTF.cpp

```diff
@@ -1638,15 +1638,28 @@ std::vector<std::byte> VTF::bake() const {
 	std::vector<std::byte> out;
 	BufferStream writer{out};
 
-	static constexpr auto writeNonLocalResource = [](BufferStream& writer_, Resource::Type type, std::span<const std::byte> data, VTF::Platform platform) {
-		if (platform != VTF::PLATFORM_PC) {
+	static auto writeNonLocalResource = [this](BufferStream& writer_, Resource::Type type, std::span<const std::byte> data, VTF::Platform platform) {
+		if (platform != VTF::PLATFORM_PC && std::endian::native == std::endian::little) {
 			BufferStream::swap_endian(reinterpret_cast<uint32_t*>(&type));
 		}
 		writer_.write<uint32_t>(type);
 		const auto resourceOffsetPos = writer_.tell();
 		writer_.seek(0, std::ios::end);
 		const auto resourceOffsetValue = writer_.tell();
-		writer_.write(data);
+
+		int8_t bpc = 8;
+		bpc = std::max(bpc, vtfpp::ImageFormatDetails::alpha(this->getFormat()));
+		bpc = std::max(bpc, vtfpp::ImageFormatDetails::red(this->getFormat()));
+		switch (bpc) {
+		case 16:
```

Comment:
Cases should get indentation, and case bodies are doubly indented from outside the switch